share/completions/gsettings.fish

```diff
@@ -0,0 +1,96 @@
+# Fish completions for glibs's "gsettings" configuration tool
+
+function __fish_complete_gsettings_args
+    set -l schema_commands get monitor writable range describe set reset reset-recursively list-keys list-children list-recursively
+    set -l key_commands get monitor writable range describe set reset
+
+    set -l schemadir
+    set -l offset 0
+    set -l cmd (commandline -poc)
+    set -e cmd[1]
+
+    if [ (count $cmd) -ge 2 ]; and string match -q -- '--schemadir' $cmd[1]
+        set schemadir --schemadir (echo $cmd[2])
+        set offset 2
+    end
+
+    if [ (count $cmd) -lt (math 1 + $offset) ]
```

Comment:
You're invoking `math` with the same arguments more than once. This screams out to be refactored to do that calculation only once, assigning the result to a local variable, then using that var.